### PR TITLE
Feature/shoes addproduct

### PR DESCRIPTION
**Pull request by balajiachanta**
Shoe add product code added

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,3 +37,5 @@ public class ClothesApiController {
 ## Github Actions
 
 - [Environment Variables](https://docs.github.com/en/actions/reference/environment-variables)
+
+shoes
```
Comment (vyomrastogi) [inline README.md]: @NagaVijaya / @balajiachanta we think this is a typo, could you check

File: src/main/java/com/cottonfactory/products/controllers/ShoeController.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class ShoeController {
+
+    @Autowired
+    private ShoeService shoeService;
+
+    @PostMapping("/api/products/shoe")
```
Comment (vyomrastogi) [inline src/main/java/com/cottonfactory/products/controllers/ShoeController.java:18]:
```suggestion
    @PostMapping("/api/v1/products/shoe")
```
the Product API home has an initial value as api/v1/products, probably a good idea to follow same versioning

File: src/test/java/com/cottonfactory/products/controllers/ShoeControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.math.BigDecimal;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
```
Comment (vyomrastogi) [inline src/test/java/com/cottonfactory/products/controllers/ShoeControllerTest.java:18]: Does it  need `@AutoConfigureTestDatabase` to validate entry to H2 during setup ?

**Review by vyomrastogi — CHANGES_REQUESTED** [review]
@NagaVijaya / @balajiachanta  could you please review the comments ?

File: src/main/java/com/cottonfactory/products/controllers/ShoeController.java (hunk 2 above)
Comment (NagaVijaya) [reply src/main/java/com/cottonfactory/products/controllers/ShoeController.java:18]: v1 is not required so not making any change

File: src/test/java/com/cottonfactory/products/controllers/ShoeControllerTest.java (hunk 3 above)
Comment (NagaVijaya) [reply src/test/java/com/cottonfactory/products/controllers/ShoeControllerTest.java:18]: testcase is working fine without @AutoConfigureTestDatabase so not making any change

File: README.md (hunk 1 above)
Comment (NagaVijaya) [reply README.md]: updated read me per comment

File: build.gradle [hunk 4]
```diff
@@ -17,9 +17,13 @@ dependencies {
...
 	runtimeOnly 'com.h2database:h2'
     testImplementation('org.springframework.boot:spring-boot-starter-test')
-
+    compileOnly 'org.projectlombok:lombok'
+    annotationProcessor 'org.projectlombok:lombok'
     compile group: 'io.springfox', name: 'springfox-swagger2', version: '2.7.0'
     compile group: 'io.springfox', name: 'springfox-swagger-ui', version: '2.7.0'
+
+    compileOnly 'org.projectlombok:lombok'
+    annotationProcessor 'org.projectlombok:lombok'
```
Comment (vyomrastogi) [inline build.gradle]:
```suggestion
    compileOnly 'org.projectlombok:lombok'
    annotationProcessor 'org.projectlombok:lombok'
    compile group: 'io.springfox', name: 'springfox-swagger2', version: '2.7.0'
    compile group: 'io.springfox', name: 'springfox-swagger-ui', version: '2.7.0'
```

Comment (vyomrastogi) [reply build.gradle]: @balajiachanta / @NagaVijaya there are duplicates for lombok

**Review by vyomrastogi — CHANGES_REQUESTED** [review]

**Review by vyomrastogi — APPROVED** [review]

**Comment by vyomrastogi** [comment]
@Protosac / @ChaoticNeutralCode  we are good with this PR

**Review by RichFromGalvanize — APPROVED** [review]
Pretty straight forward additions here.

**Pull request merged**